envs/Pendulum/Pendulum.py

```diff
@@ -10,11 +10,48 @@
 from gymnasium.envs.classic_control import utils
 from gymnasium.error import DependencyNotInstalled
 
+import tensorflow as tf
+from cmorl.utils.loss_composition import p_mean
+from cmorl.utils.reward_utils import CMORL, RewardFnType
+
 
 DEFAULT_X = np.pi
 DEFAULT_Y = 1.0
 
 
+def multi_dim_reward(state, action, env: "PendulumEnv"):
+    u = action
+    # x, y, thdot = state  # th := theta
+    # th = np.arctan2(y, x)
+    th, thdot = state  # th := theta
+    angle_rw = 1.0 - normed_angular_distance(th, env.setpoint)
+
+    # Normalizing the torque to be in the range [0, 1]
+    normalized_u = u / env.max_torque
+    normalized_u = abs(normalized_u)
+    actuation_rw = 1.0 - normalized_u
+    # check if actuation_rw is an array or a scalar
+    if isinstance(u, np.ndarray):
+        actuation_rw = actuation_rw[0]
+    # Merge the angle reward and the normalized torque into a single reward vector
+    rw_vec = np.array([angle_rw, actuation_rw], dtype=np.float32)
+    return rw_vec
+
+
+def composed_reward_fn(state, action, env: "PendulumEnv"):
+    return multi_dim_reward(state, action, env)[0:1]
+
+
+@tf.function
+def q_composer(q_values):
+    q1_c = q_values[0] ** 2
+    q2_c = q_values[1]
```

Comment:
We can also reduce importance by doing something like 0.5 + 0.5*q_values[1], that makes it closer to a curriculum where you start with optimizing q1 then q2. Also because this is inside the environment I would give the Q values names for example, q_actuation and q_angle. Also hopefully we can use the DFL stuff soon so these things are just a bit nicer to work with